sigma/plugins/roles/setcolor.py

```diff
@@ -0,0 +1,62 @@
+import discord
+from sigma.core.permission import check_man_roles
+from sigma.core.rolecheck import matching_role
+
+async def setcolor(cmd, message, args):
+    # if no args, send help message
+    if not args:
+        await message.channel.send(cmd.help())
+        return
+
+    # checks permissions
+    if not check_man_roles(message.author, message.channel):
+        out_content = discord.Embed(type='rich', color=0xDB0000, title='⛔ Error')
+        out_content.add_field(name='Insufficient Permissions', value='You require the Manage Roles permission.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    color = args[0].lstrip('#').upper()
+    # check if color is a proper format
+    if not is_valid_hex(color):
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='⚠ Error')
+        out_content.add_field(name='Invalid Input', value='A valid hexadecimal color code has numbers from 0–9 and letters from A–F, and has six characters.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check if role exists
+    exists = matching_role(message.guild, color)
+    if not exists:
+        discord_colour = discord.Colour(int(color, 16))
+        target_role = await message.guild.create_role(name=color, colour=discord_colour, reason="Color role")
+    else:
+        target_role = exists
+
+    target = message.author if not message.mentions else message.mentions[0]
+
+    # get user's roles, remove old role, append new role, and set user's roles
+    list_of_roles = target.roles
+
+    for index, role in enumerate(list_of_roles):
```

Comment:
see https://stackoverflow.com/a/52458154/3256177, except try `name` instead of `id`

Reply:
I can't use this, because of Line 40: `if is_valid_hex(role.name):`. Here, I'm checking if the name passes the validity test. The `discord.utils.get` function is good for finding a role that you know the name of, but we don't know the name of the role. Unless you want to try all 16 777 216 values that a hex code can be.

Reply:
ok